### PR TITLE
HSEARCH-4287 + HSEARCH-4288 + HSEARCH-4289 Outbox processing order and database locking improvements

### PR DESCRIPTION
**Pull request by yrodiere**
* [HSEARCH-4289](https://hibernate.atlassian.net/browse/HSEARCH-4289): Ignore database-locked events in outbox automatic indexing processors
* [HSEARCH-4288](https://hibernate.atlassian.net/browse/HSEARCH-4288): Resume testing against MS SQL Server in CI
* [HSEARCH-4287](https://hibernate.atlassian.net/browse/HSEARCH-4287): Make processing of outbox events indifferent to processing order

### CODE REVIEW

**Comment by yrodiere** [comment]
Build with tests for the main databases: https://ci.hibernate.org/blue/organizations/jenkins/hibernate-search/detail/PR-2641/2/pipeline

**Comment by yrodiere** [comment]
I pushed a commit to fix the problem with MariaDB.

New build with tests for the main databases: https://ci.hibernate.org/blue/organizations/jenkins/hibernate-search/detail/PR-2641/7/pipeline

File: mapper/orm/src/main/java/org/hibernate/search/mapper/orm/coordination/databasepolling/impl/OutboxEventBackgroundProcessor.java [hunk 1]
```diff
@@ -106,7 +105,19 @@ public HibernateOrmOutboxWorker(SessionFactoryImplementor sessionFactory) {
...
+					}
+					catch (OptimisticLockException lockException) {
+						// Don't be fooled by the exception type, this is actually a *pessimistic* lock failure.
+						// It can happen with some databases (Mariadb before 10.6, perhaps others) that do not support
+						// skipping locked rows (see LockOptions.SKIP).
+						// If that happens, we will just log something and try again later.
+						// See also https://jira.mariadb.org/browse/MDEV-13115
+						log.outboxEventProcessorUnableToLock( name, lockException );
+						events = Collections.emptyList();
+					}
```
Comment (fax4ever) [inline mapper/orm/src/main/java/org/hibernate/search/mapper/orm/coordination/databasepolling/impl/OutboxEventBackgroundProcessor.java:120]: Maybe we should link the issue HSEARCH-4297

**Review by fax4ever — APPROVED** [review]
Good job! I have nothing to say of it.
Just few comments on style.

**Comment by yrodiere** [comment]
Fixed the style issues and merged. Thanks!

**Pull request merged**